### PR TITLE
[SQL] Implement the NOW table as a generator

### LINKED ISSUE #2235: [SQL] [adapter] Create an adapter to populate the NOW table
The adapter has to insert a new TIMESTAMP value every step.
It should not delete the previous value.

### PR DESCRIPTION
**Pull request by mihaibudiu**
Fixes #2235

### CODE REVIEW

File: docs/sql/datetime.md [hunk 1]
```diff
@@ -297,36 +297,16 @@ computation on whole days.
...
-
-This table does not currently get populated automatically.  Instead,
-the user is responsible for supplying the data to this table.  In
-every step of the circuit the user has to insert a new value in this
-table, which should be larger than the previous value.
+| Operation     | Description         | Example                        |
+|---------------|---------------------|--------------------------------|
+| `NOW`         | Returns a timestamp | `NOW()` => 2024-07-10 00:00:00 |
 
 :::warning
```
Comment (lalithsuresh) [inline docs/sql/datetime.md:304]: I think you need something to close the warning. Right now, all content after it shows up as a warning.

File: docs/sql/datetime.md [hunk 2]
```diff
@@ -297,36 +297,16 @@ computation on whole days.
...
 
 Programs that use `NOW()` can be very inefficient.  For example, a
```
Comment (lalithsuresh) [inline docs/sql/datetime.md:306]: Are there programs using now() that will be efficient though? E.g., if it's just a projection that adds a column in a view which is then used for an ASOF join or something?

Comment (mihaibudiu) [reply docs/sql/datetime.md:306]: that's what I am fixing now

File: docs/sql/datetime.md (hunk 1 above)
Comment (mihaibudiu) [reply docs/sql/datetime.md:304]: I thought I did. IT's just :::

Comment (lalithsuresh) [reply docs/sql/datetime.md:304]: Ah, if you no longer see it when viewing with `yarn start` then it's fine.

Comment (mihaibudiu) [reply docs/sql/datetime.md:304]: it may have gotten lost in the edits when I moved this text to a different file.
I will add it back before I merge.

**Pull request merged**